Classes/Connection/Elasticsearch/IndexFactory.php

```diff
@@ -41,19 +56,25 @@ class IndexFactory implements Singleton
      */
     public function getIndex(Connection $connection, $documentType)
     {
-        // TODO: Fetch index name from configuration, based on $documentType.
         $index = $connection->getClient()->getIndex('typo3content');
 
-        try {
-            // TODO: Provide configuration?!
-            // http://elastica.io/getting-started/storing-and-indexing-documents.html#section-analysis
-            $index->create();
-        } catch (ResponseException $exception) {
-            if (stripos($exception->getMessage(), 'already exists') === false) {
-                throw $exception;
-            }
+        if ($index->exists() === false) {
+            $index->create($this->getConfigurationFor($documentType));
         }
 
         return $index;
     }
+
+    /**
+     * @param string $documentType
+     * @return array
+     */
+    protected function getConfigurationFor($documentType)
+    {
+        try {
+            return $this->configuration->get('indexing.' . $documentType . '.index');
```

Comment:
We can not just pipe the configuration as TypoScript is incompatible. I've to adjust this and explode comma separated values for `filter` and `char_filter`.

They have to be arrays without keys: https://www.elastic.co/guide/en/elasticsearch/reference/5.2/analysis-custom-analyzer.html#_example_configuration_5